src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputIfElse.java

```diff
@@ -0,0 +1,31 @@
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+public class InputIfElse {
+
+    boolean condition() {
+        return false;
+    }
+
+    void testIfElse() {
+        //Valid
+        if (condition()) {
+            testIfElse();
+        } else {
+            testIfElse();
+        }
+
+        //Invalid
+        if(condition()) {
```

Comment:
please put "//violation" as training comment. to easily catch lines where you expect violation.
Please do the same for other input code.